source/frontend-docs.rst

```diff
@@ -0,0 +1,218 @@
+    * ``@param`` tags each expected parameter, including:
+        * name
+        * type
+        * prose description
+    * For Object-type parameters, include separate ``@param`` entries for each
+      property expected on the Object
+    * ``@returns`` tag giving the function's return value, including:
+        * type
+        * prose description
+    * ``@fires`` tags indicating events fired within the function body
+* If a function is a callback:
+    * Declare its doc comment with ``@callback``
+    * Specify the event(s) it's attached to with ``@listens``
+* Include ``@see`` references wherever another part of the code provides useful
+  context for interpreting the comment.
+
+For the full details of what JSDoc is capable of, see `the JSDoc website <http://usejsdoc.org/>`_.
+
+Examples
+========
+
+Comment style
+-------------
+
+An ES6 module can be documented by adding a JSDoc comment to its
+constructor. This comment includes a brief prose description of
+the class's nature; it explicitly identifies the function as a
+constructor using ``@constructs``; it spells out the type of its single
+required parameter, ``options``, and the attribute ``id`` that it
+requires; and it uses ``@see`` to link to the docs for the method
+``appSetup`` which does the constructor's heavy lifting::
+
+    export default class ShowMatches extends Model {
+      /**
+       * Create a new app state model.
+       *
+       * @constructs ShowMatches
+       * @param {Object} options - initialization options for app.
+       * @param {string} options.id - ID of dataset.
+       *
+       * @see ShowMatches#appSetup
+       */
+      constructor (options, ...rest) {
+        super(options, ...rest);
+        this.__options__ = options;
+        this.appSetup(options);
+      }
+
+      // ...
+    }
+
+This example of a JSDoc comment for a method on this class contains
+a prose description indicating how it fits in with the flow of the
+app; it specifies the type of its single parameter by linking to
+the docs for a class defined in a separate module, also indicating
+with the ``Type[]`` notation that the argument is an array of values
+of that type; it indicates via ``@fires`` that calling the function
+fires an event, and it links to the docs for that event; and it
+uses ``@see`` to accompany the reference to ``PossibleMatches`` with
+an explicit link to its documentation::
+
+    /**
+     * Method called to compare a collection of potential matches, triggered
+     * by the PossibleMatches view.
+     *
+     * @method ShowMatches#compareMatches
+     * @param {module:models/PossibleMatch.PossibleMatch[]} matches - set of
+     *     matches to compare
+     * @fires ShowMatches#event:change:comparing_matches
+     * @see module:views/PossibleMatches
+     */
+    compareMatches (models) {
+      this.set('comparing_matches', new PossibleMatchesCollection(models));
+    }
+
+**Important**: note that ``@method`` gives the name of the class and the
```

Comment:
Sphinx has nice built-in support for formatting warning boxes so you don't have to format them yourself - see "Admonitions" here: http://www.sphinx-doc.org/en/stable/rest.html#directives